altair/utils/core.py

```diff
@@ -240,21 +240,113 @@ def sanitize_geo_interface(geo):
     return geo
 
 
+def sanitize_series(s):
```

Comment:
Could the goal of this change be accomplished by leaving ``sanitize_dataframe`` untouched, and adding this?
```
def sanitize_series(s):
  return sanitize_dataframe(s.to_frame())
```

Reply:
Or perhaps ``sanitize_dataframe(s.reset_index())`` since visualization of series data rarely makes sense without its index?

Reply:
Yeah, that would be doable.

I refactored `sanitize_dataframe` because for someone coming into the codebase fresh it would make more sense have `sanitize_dataframe` depend on `sanitize_series` than the other way around. Let me know if you still prefer for me to change it.

In the context of `reset_index`, I think that if it is done for series, it should also be done for dataframes to make sure things stay consistent.

Reply:
I disagree on defaulting to reset_index. Size of serialized data is a concern in Alatir; not serializing the index by default allows people to choose whether they want the index included (by using `reset_index` or not). If we reset the index by default, there is no way to opt-out of including an extra column in every data source derived from a dataframe.

Reply:
Looking at it from my daily work, I am pretty much *always* typing `reset_index()` because my indices have meaningful information about the data such as the iteration number, the year, or a personal identifier that I want to include in my visualizations *and* use as indices.  
I understand the need to be efficient with respect to the size of serialized data. By adding `reset_index`, worst case scenario is formed with an inline dataset with 1 column and many rows that gets repeated a lot. But (at least for me) it is quite rare to have very few columns and the relative damage is a lot less with more columns. If the user is that close to the limits of inline data, they can easily start using the json transformer.
I should note I always use the json transformer so maybe I'm not feeling the pain of serialization issues.

With regards to the comment about series support, I think that's a point well-taken. I took Series because it seemed to be the closest one to DataFrame and thus the easiest to implement by mimicking what is done for DataFrame.

Reply:
I don't see any downside to Altair supporting series input by calling `to_frame()` to convert it to a dataframe. I think that would be a convenient feature.

Reply:
Regarding the JSON data transformer: it doesn't work consistently across different display frontends, so I've stopped recommending it as a general solution to larger datasets.

Reply:
I've rolled back most of the sanitize series support, so it's now a simple call to `reset_index()`. Of course I'll also add the docs for it. The weird thing about it still is that you need to explicitly state the encoding type even though pandas can infer the dtypes of a series easily.

I also had a look at dictionaries. Do you think it's worth to include the cast to a Dataframe there as well (simply calling `pd.DataFrame` on the underlying dict)? Or to take a list of records and adding the `{"values": <stuff>}` bit automatically, or casting that list to a DataFrame as well?  

Additionally I discovered that the error for a dict not including a "values" key (in the `to_values` transformer) will never get raised: first schema validation fails, and if validation is turned off, the `_prepare_data` function in `api.py` doesn't route dicts through the transformers.

In regards to the JSON data transformer, any idea as to potential universal solutions? Altair server?

Reply:
Altair server is a good option, except when it isn't :grin: (for example, when using a jupyterhub with no port forwarding).

Reply:
Any comments on the dict and list ideas? If not, I'll take them out and complete the PR with just the series functionality.